pcdsdevices/dg645.py

```diff
@@ -0,0 +1,447 @@
+        
+            
+    def tweak_offset(self, value=None, inc=False, dec=False):
+        """
+        If value is None:
+        Returns tweak offset value on DG645 channel
+        If value is not None:
+        Sets tweak delay value to set value 
+        To tweak increase: tweak(inc=True)
+        To tweak decrease: tweak(dec=True)
+        """
+        if value is None:
+            return self.off_Tweak.get()
+        else:
+            checkValue(value, self.off_low_lim, self.off_high_lim)
+            self.off_Tweak.put(value)
+            
+        if inc:
+            self.off_Inc.put(1)
+        if dec:
+            self.off_Dec.put(1)
+        
+
+    def mvr_off(self, delta):
+        """
+        Moves the pulse offset by delta relative to the current offset.
+        """
+        self.offset = delta + self.offset
+        
+        
+    @property
+    def amplitude(self):
+        """
+        If value is None:
+        Returns the current offset value on the DG645 channel
+        If valueis not  None:
+        The offset value on the DG645 channel is changed to that
+        value.        
+        """
+        return self.amp_AI.get()
+            
+    @amplitude.setter
+    def amplitude(self, value):
+        checkValue(value, self.amp_low_lim, self.amp_high_lim)
+        self.amp_AO.put(value)
+            
+        
+    def tweak_amplitude(self, value=None, inc=False, dec=False):
+        """
+        If valueis None:
+        Returns tweak amplitude value on DG645 channel
+        If value is not None:
+        Sets tweak amplitude value to set value 
+        To tweak increase: tweak(inc=True)
+        To tweak decrease: tweak(dec=True)
+        """
+        if value is None:
+            return self.amp_Tweak.get()
+        else:
+            checkValue(value, self.amp_low_lim, self.amp_high_lim)
+            self.amp_Tweak.put(value)
+            
+        if inc:
+            self.amp_Inc.put(1)
+        if dec:
+            self.amp_Dec.put(1)
+                
+
+    def mvr_amp(self, delta):
+        """
+        Moves the pulse amplitude by delta relative to the current amplitude.
+        """
+        self.amplitude = delta + self.amplitude
+
+
+
+class trigger(Device):
+    """
+    Class that defines an output (pulse) channel of a DG645 delay generator.
+    
+    Parameters
+    ---------
+    prefix : ``str``
+        The PV base of the relevant DG, i.e 'MEC:LAS:DDG:01:'
+    
+    name : ``str``                                                               
+        Alias for the delay generator triggers
+    """
+    trig_sourceMO  = Cpt(EpicsSignal, 'triggerSourceMO')
+    trig_sourceMI  = Cpt(EpicsSignalRO, 'triggerSourceMI')
+    trig_inhibMO   = Cpt(EpicsSignal, 'triggerInhibitMO')
+    trig_inhibMI   = Cpt(EpicsSignalRO, 'triggerInhibitMI')
+    
+    def __init__(self, prefix, name='triggers', **kwargs):
+        self.__sourceDict = {"Internal":0, "Ext ^edge":1, "Ext ~edge":2, 
```

Comment:
Are these strings coming from EPICS directly? If unsure, do you see the following when debugging with `caget`:
```bash
$ caget ...triggerSourceMO
0
$ caget -S ...triggerSourceMO
Internal
```
If that's the case, you can mark the `EpicsSignal` components above with `string=True`, and it will handle the enum conversion automatically.

If not, you could leave it as-is and invert the dictionaries automatically, at least:
```python
self.__sourceDictInv = dict((value, key) for key, value in self.__sourceDict.items())
```